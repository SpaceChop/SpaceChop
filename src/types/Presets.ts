import * as t from 'runtypes';
import PresetConfig from './PresetConfig';

const Presets = t.Dictionary(PresetConfig, 'string');

export type Presets = t.Static<typeof Presets>;
export default Presets;
