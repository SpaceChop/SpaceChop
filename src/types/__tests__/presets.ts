import Presets from '../Presets';

describe('validate Presets', () => {
  describe('invalid', () => {
    it('should require presets', () => {
      const presets = null;
      expect(() => Presets.check(presets)).toThrowErrorMatchingSnapshot();
    });

    it('should throw on invalid presets', () => {
      let presets = {
        t_200: {},
      };
      expect(() => Presets.check(presets)).toThrowErrorMatchingSnapshot();

      presets = {
        t_200: {
          steps: {},
        },
      };
      expect(() => Presets.check(presets)).toThrowErrorMatchingSnapshot();

      presets = {
        t_200: {
          steps: [{}],
        },
      };
      expect(() => Presets.check(presets)).toThrowErrorMatchingSnapshot();
    });
  });
});
