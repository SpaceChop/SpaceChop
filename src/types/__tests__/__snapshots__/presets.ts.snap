// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Presets invalid should require presets 1`] = `"Expected { [_: string]: { steps: { $compress?: { quality?: number; lossy?: boolean; }; $crop?: { width?: number | { from_path: string; }; height?: number | { from_path: string; }; gravity?: (\\"center\\" | \\"face\\" | \\"north\\" | \\"northeast\\" | \\"east\\" | \\"southeast\\" | \\"south\\" | \\"southwest\\" | \\"west\\" | \\"northwest\\") | { from_path: string; }; }; $fill?: { width: number | { from_path: string; }; height: number | { from_path: string; }; } & { gravity?: (\\"center\\" | \\"face\\" | \\"north\\" | \\"northeast\\" | \\"east\\" | \\"southeast\\" | \\"south\\" | \\"southwest\\" | \\"west\\" | \\"northwest\\") | { from_path: string; }; }; $fit?: { width?: number | { from_path: string; }; height?: number | { from_path: string; }; }; $format?: { type: (\\"jpeg\\" | \\"png\\" | \\"gif\\" | \\"webp\\") | { from_path: string; }; }; $resize?: { width: number | { from_path: string; }; height: number | { from_path: string; }; }; $strip?: { icc_profile?: boolean; }; }[]; } }, but was null"`;

exports[`validate Presets invalid should throw on invalid presets 1`] = `"Expected array, but was undefined"`;

exports[`validate Presets invalid should throw on invalid presets 2`] = `"Expected array, but was object"`;

exports[`validate Presets invalid should throw on invalid presets 3`] = `"please no empty step"`;
